### PR TITLE
CMake: use CMAKE_INSTALL_LIBDIR in INSTALL_PLUGIN_DIR

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: gdal.cmake [hunk 1]
```diff
@@ -353,7 +353,7 @@ if (GDAL_ENABLE_MACOSX_FRAMEWORK)
 else ()
   include(GNUInstallDirs)
   set(INSTALL_PLUGIN_DIR
-      "lib/gdalplugins"
+      "lib${LIB_SUFFIX}/gdalplugins"
```
Comment (dg0yt) [inline gdal.cmake:356]: With `GNUInstallDirs` ahead, shouldn't this be:
~~~cmake
   "${CMAKE_INSTALL_LIBDIR}/gdal_plugins"
~~~

Comment (rouault) [reply gdal.cmake:356]: @manisandro opinion w.r.t above suggestion of using ${CMAKE_INSTALL_LIBDIR} ?

Comment (manisandro) [reply gdal.cmake:356]: Works for me!

Comment (rouault) [reply gdal.cmake]: PR amended

**Pull request merged**